src/jl_uv.c

```diff
@@ -174,9 +179,12 @@ static void jl_uv_closeHandle(uv_handle_t *handle)
         ct->world_age = last_age;
         return;
     }
-    if (handle == (uv_handle_t*)&signal_async || handle == (uv_handle_t*)&wait_empty_worker)
+    if (handle == (uv_handle_t*)&wait_empty_worker)
+        handle->type = 0;
```

Comment:
```suggestion
        handle->type = UV_UNKNOWN_HANDLE;
```
The implicit conversion here breaks building on musl Linux.